### PR TITLE
Dev branch c

### PR DESCRIPTION
**Pull request by mitalraithatha**
Task C is ready for review.
.
receiver_cadmium.hpp
receiver->main.cpp
By Mital

sender_cadmium.hpp
subnet->main.cpp
by Sreejith

### CODE REVIEW

File: include/atomics/receiver_cadmium.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        			}
+        			for(const auto &x : get_messages<typename defs::input>(mbs)) {
+        				state.acknowledgement_number = static_cast < int > (x.value);
+        				state.sending = true;
+        			}
+                           
+        }
+        /**
+		* confluence transition calls internal_transition and external_transition respectively.
+		* @param e TIME type variable
```
Comment (tarimislam) [inline include/atomics/receiver_cadmium.hpp:109]: Can you add @param[in] instead @param as the professor said

**Review by tarimislam — CHANGES_REQUESTED** [review]

File: test/src/receiver/main.cpp [hunk 2]
```diff
@@ -1,130 +1,185 @@
...
+class ApplicationGen : public iestream_input<message_t,T> {
+	public:
+		/**
+		 * Default constructor for class application generator.
+		*/ 
+		ApplicationGen() = default;
+		/**
+		 * Parameterized constructor for class application generator.
+		 * This constructor ask for file path which contain inputs.
+		 * @param full file path
```
Comment (tarimislam) [inline test/src/receiver/main.cpp:60]: Add @param[in]

**Review by tarimislam — CHANGES_REQUESTED** [review]

File: include/atomics/receiver_cadmium.hpp (hunk 1 above)
Comment (mitalraithatha) [reply include/atomics/receiver_cadmium.hpp:109]: I think this is not mandatory thing.
I follow the doxygen manual format and in that also this is not compulsory.

**Review by tarimislam — APPROVED** [review]
Everything looks ok

**Review by navid-hossain — APPROVED** [review]
Everything looks good

**Pull request merged**